cmd/secret-wrapper/main.go

```diff
@@ -172,6 +172,18 @@ func execCmd(argv []string) error {
 	proc := exec.Command(argv[0], argv[1:]...)
 	proc.Stdout = os.Stdout
 	proc.Stderr = os.Stderr
+	// provide a writeable home so kubectl discovery can be cached
+	home, set := os.LookupEnv("HOME")
+	needHome := !set
+	if set {
+		if err := syscall.Access(home, syscall.O_RDWR); err != nil {
+			// $HOME is set but not writeable
+			needHome = true
+		}
+	}
+	if needHome {
+		proc.Env = append(os.Environ(), "HOME=/alabama")
+	}
```

Comment:
This function will get quite involved, specially with the following PR.  Can we leave only the execution bits here and move the new code to a separate one?  Adding the environment as a parameter looks like a good interface.

Reply:
Sure